### PR TITLE
chore(all): integrate codecov

### PR DESCRIPTION
**Pull request by cyberhorsey**

### CODE REVIEW

**Review by dionysuzx — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 [![Solidity](https://github.com/taikochain/taiko-mono/actions/workflows/solidity.yml/badge.svg)](https://github.com/taikochain/taiko-mono/actions/workflows/solidity.yml)
 
+[![codecov](https://codecov.io/gh/taikochain/taiko-mono/branch/main/graph/badge.svg?token=E468X2PTJC)](https://codecov.io/gh/taikochain/taiko-mono)
```
Comment (davidtaikocha) [inline README.md:5]: Shall we use different badges for different flags ( one for contract and another one for relayer )? https://community.codecov.com/t/badges-for-flags/270/3

File: .github/workflows/solidity.yml [hunk 2]
```diff
@@ -59,3 +59,9 @@ jobs:
       - name: protocol - Deploy L1 Contracts
         working-directory: ./packages/protocol
         run: pnpm deploy:hardhat
+
+      - name: relayer - Upload coverage to Codecov
```
Comment (davidtaikocha) [inline .github/workflows/solidity.yml:63]:
```suggestion
      - name: protocol - Upload coverage to Codecov
```

File: README.md (hunk 1 above)
Comment (cyberhorsey) [reply README.md]: yes, lets! labels `Relayer` and `Protocol `?

Comment (davidtaikocha) [reply README.md]: Sounds good

**Review by davidtaikocha — DISMISSED** [review]

File: .github/workflows/solidity.yml (hunk 2 above)
Comment (davidtaikocha) [reply .github/workflows/solidity.yml:63]: Looks like this is still unresolved?

**Review by davidtaikocha — APPROVED** [review]

File: .github/workflows/solidity.yml (hunk 2 above)
Comment (cyberhorsey) [reply .github/workflows/solidity.yml]: ah, you can feel free to click "Commit suggestion" on these if youd like for things like this in the future : )

Done!

**Review by dantaik — APPROVED** [review]

**Pull request merged**